src/en/darthsdroids/src/eu/kanade/tachiyomi/extension/en/darthsdroids/DarthsDroids.kt

```diff
@@ -0,0 +1,218 @@
+        thumbnail_url = dndThumbnailUrlForTitle(nthManga)
+        title = mangaTitle
+        author = "David Morgan-Mar & Co."
+        artist = "David Morgan-Mar & Co."
+        description = """What if Star Wars as we know it didn't exist, but instead the
+            |plot of the movies was being made up on the spot by players of
+            |a Tabletop Game?
+            |
+            |Well, for one, the results might actually make a lot more sense,
+            |from an out-of-story point of view…
+        """.trimMargin()
+        genre = "Campaign Comic, Comedy, Space Opera, Science Fiction"
+        status = mangaStatus
+        update_strategy = when (mangaStatus) {
+            SManga.COMPLETED -> UpdateStrategy.ONLY_FETCH_ONCE
+            else -> UpdateStrategy.ALWAYS_UPDATE
+        }
+        initialized = true
+    }
+
+    override fun popularMangaRequest(page: Int): Request =
+        GET("$baseUrl/archive.html", headers)
+
+    override fun popularMangaParse(response: Response): MangasPage {
+        val mainArchive = response.asJsoup()
+        val archiveData = mainArchive.select("div.text > table.text > tbody > tr")
+
+        val mangas = mutableListOf<SManga>()
+        var nextMangaTitle = name
+        var nthManga = 0
+
+        run stop@{
+            archiveData.forEach {
+                val maybeTitle = it.selectFirst("th")?.text()
+                if (maybeTitle != null) {
+                    nextMangaTitle = "$name $maybeTitle"
+                } else {
+                    val maybeArchive = it.selectFirst("""td[colspan="3"] > a""")?.attr("href")
+                    if (maybeArchive != null) {
+                        mangas.add(dndManga(maybeArchive, nextMangaTitle, SManga.COMPLETED, nthManga++))
+                    } else {
+                        // We reached the end, assuming the page layout stays consistent beyond D&D8.
+                        // Thus, we append our final manga with this current page as its archive.
+                        // Unfortunately this means we will needlessly fetch this page twice.
+                        mangas.add(dndManga("/archive.html", nextMangaTitle, SManga.ONGOING, nthManga))
+                        return@stop
+                    }
+                }
+            }
+        }
+
+        return MangasPage(mangas, false)
+    }
+
+    // Not efficient, but the simplest way for me to refresh.
+    // We also can’t really use the `mangaDetailsRequest + mangaDetailsParse`
+    // approach, for we actually expect one of the books’ `url`s to change.
+    override fun fetchMangaDetails(manga: SManga): Observable<SManga> =
+        fetchPopularManga(0)
+            .map { mangasPage ->
+                mangasPage
+                    .mangas
+                    // Do not test for URL-equality, for the last book will always
+                    // eventually migrate its archive page from `/archive.html` to
+                    // its own page.
```

Comment:
can you clarify? manga's url can't be changed once set

Reply:
It’s a weird site. Their main archive is `/archive.html`. But that archive is a combined feed of previously released and completed books, as well as *all* pages of the currently releasing book. Once the current book is done, all the pages disappear from that feed, the last book gets its own complete archive page (in this case likely `/archive8.html`), and `/archive.html` is repurposed for their next book (likely Star Wars 9), until the end.

Reply:
I documented this weirdness, so that other devs won’t be as surprised when picking up this extension or trying to learn from it.